internal/util/http.go

```diff
@@ -25,6 +25,17 @@ import (
 	"github.com/coreos/ignition/internal/version"
 )
 
+// A timeout of 0 indicates no timeout
+var (
+	httpResponseHeaderTimeout int
```

Comment:
Pass them as arguments to `NewHttpClient`. If we are calling that too many times, maybe we should reuse HttpClients a bit more.